internal/cmdsearch/searchreplace_test.go

```diff
@@ -293,8 +293,38 @@ metadata:
  `,
 	},
 	{
-		name: "search by array objects path",
-		args: []string{"--by-path", "spec.foo[1].c"},
+		name: "search replace by array path regex",
+		args: []string{"--by-path", "spec.foo[1]", "--put-literal", "c"},
```

Comment:
What happens if the provided number is outside the bounds of the array?

Reply:
It's a no match. Similar to providing non-existing path. We will support appending elements to arrays in future. https://mikefarah.gitbook.io/yq/usage/path-expressions#appending-to-arrays

Reply:
This seems like something that would be worthy of testing. This also includes the use of negative indexes (as described in the link) if that is something that we support.

Reply:
Added a [test case](https://github.com/GoogleContainerTools/kpt/pull/1225/files#diff-a0a456fc7bee0fb4aadca84d28d1dd6444127841d2b9499262d1e09cae3b2945R326). We may add more features based on the requirements in future. yq supports negative indices but for us it depends on use-cases we might encounter in the future.